QuizApp/QuizApp/Storage/UserDatabaseDataSource.swift

```diff
@@ -18,8 +32,38 @@ class UserDatabaseDataSource: UserDatabaseDataSourceProtocol {
         secureStorage.save(accessToken: accessToken)
     }
 
+    func deleteAccessToken() {
+        secureStorage.deleteAccessToken()
+    }
+
+    func deleteUserInfo() {
+        userDefaults.set("", forKey: usernameKey)
+    }
+
+    func save(userInfo: UserInfoDataSourceModel) {
+        userDefaults.set(userInfo.username, forKey: usernameKey)
+    }
+
     var accessToken: String? {
         secureStorage.accessToken
     }
 
+    var userInfo: UserInfoDataSourceModel {
+        UserInfoDataSourceModel(username: userDefaults.string(forKey: usernameKey) ?? "")
+    }
+
+}
+
+struct UserInfoDataSourceModel {
+
+    let username: String
+
+}
+
+extension UserInfoDataSourceModel {
+
+    init(_ userInfo: UserInfoModel) {
```

Comment:
DS should not have "knowledge" of UC classes but vice-versa is ok. So, this extension should be in UC class.